packages/legend-studio/src/stores/editor-state/element-editor-state/connection/DatabaseBuilderState.ts

```diff
@@ -0,0 +1,602 @@
+import type { GeneratorFn } from '@finos/legend-studio-shared';
+import {
+  addUniqueEntry,
+  assertNonEmptyString,
+  assertTrue,
+  guaranteeNonNullable,
+  isNonNullable,
+} from '@finos/legend-studio-shared';
+import { observable, action, makeObservable, flow } from 'mobx';
+import {
+  DatabaseBuilderInput,
+  StorePattern,
+  TargetDatabase,
+} from '../../../../models/metamodels/pure/action/generation/DatabaseBuilderInput';
+import { PackageableElementExplicitReference } from '../../../../models/metamodels/pure/model/packageableElements/PackageableElementReference';
+import type { RelationalDatabaseConnection } from '../../../../models/metamodels/pure/model/packageableElements/store/relational/connection/RelationalDatabaseConnection';
+import type { Column } from '../../../../models/metamodels/pure/model/packageableElements/store/relational/model/Column';
+import { Database } from '../../../../models/metamodels/pure/model/packageableElements/store/relational/model/Database';
+import type { Schema } from '../../../../models/metamodels/pure/model/packageableElements/store/relational/model/Schema';
+import type { Table } from '../../../../models/metamodels/pure/model/packageableElements/store/relational/model/Table';
+import {
+  isValidFullPath,
+  resolvePackageNameAndElementName,
+} from '../../../../models/MetaModelUtility';
+import type { Entity } from '../../../../models/sdlc/models/entity/Entity';
+import { CORE_LOG_EVENT } from '../../../../utils/Logger';
+import type { EditorStore } from '../../../EditorStore';
+
+export abstract class DatabaseBuilderTreeNodeData implements TreeNodeData {
+  isOpen?: boolean;
+  id: string;
+  label: string;
+  parentId?: string;
+  childrenIds?: string[];
+  isChecked = false;
+
+  constructor(id: string, label: string, parentId: string | undefined) {
+    this.id = id;
+    this.label = label;
+    this.parentId = parentId;
+  }
+}
+
+export class SchemaDatabaseBuilderTreeNodeData extends DatabaseBuilderTreeNodeData {
+  schema: Schema;
+
+  constructor(id: string, parentId: string | undefined, schema: Schema) {
+    super(id, schema.name, parentId);
+    this.schema = schema;
+  }
+}
+
+export class TableDatabaseBuilderTreeNodeData extends DatabaseBuilderTreeNodeData {
+  table: Table;
+  constructor(id: string, parentId: string | undefined, table: Table) {
+    super(id, table.name, parentId);
+    this.table = table;
+  }
+}
+
+export class ColumnDatabaseBuilderTreeNodeData extends DatabaseBuilderTreeNodeData {
+  column: Column;
+  constructor(id: string, parentId: string | undefined, column: Column) {
+    super(id, column.name, parentId);
+    this.column = column;
+  }
+}
+
+export interface DatabaseBuilderTreeData
+  extends TreeData<DatabaseBuilderTreeNodeData> {
+  database: Database;
+}
+
+export class DatabaseBuilderState {
+  editorStore: EditorStore;
+  connection: RelationalDatabaseConnection;
+  modal = false;
+  databaseGrammar = '';
```

Comment:
```suggestion
  databaseGrammarCode = '';
  // OR
  databaseGrammarText = '';
```

Reply:
> 
should be a commit not a commit suggestion